### ISSUE TITLE
Crash on print buffer (cmd+p)

### ISSUE THREAD

**Issue by p666** [syl20bnr/spacemacs#5172]
#### Description

Emacs crashes on cmd+p (print buffer)
#### Reproduction guide
- Start Emacs
- cmd+p
- esc to close the popup window
- emacs crashes

_Observed behaviour:_
emacs crashes

_Expected behaviour:_
Close popup window
#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.11
- Spacemacs branch: nil (rev. 93e55bb)
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:

``` elisp
(osx shell emacs-lisp git org themes-megapack theming c-c  )
```
#### Backtrace

```
BACKTRACE IF RELEVANT
```

**Comment by syl20bnr** [comment]
Can you reproduce it by restarting Emacs with `SPC q D` ? (It restarts Emacs with `emacs -Q` with optional package)

**Comment by p666** [comment]
After restarting using `SPC q D` (without any package) I couldn't reproduce.
Here is an os crash info:

Process:               Emacs-x86_64-10_9 [21678]
Path:                  /Applications/Emacs.app/Contents/MacOS/Emacs-x86_64-10_9
Identifier:            org.gnu.Emacs
Version:               Version 24.5 (9.0)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Emacs-x86_64-10_9 [21678]
User ID:               501

Date/Time:             2016-02-21 22:06:03.960 +0200
OS Version:            Mac OS X 10.11.3 (15D21)
Report Version:        11
Anonymous UUID:        BEC23A94-B250-51D0-7CF2-B925270B8373


Time Awake Since Boot: 2100000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGABRT)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000010
Exception Note:        EXC_CORPSE_NOTIFY

VM Regions Near 0x10:
--> 
    __TEXT                 0000000100000000-000000010020e000 [ 2104K] r-x/rwx SM=COW  /Applications/Emacs.app/Contents/MacOS/Emacs-x86_64-10_9

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff96f1d9fe __kill + 10
1   Emacs-x86_64-10_9             	0x00000001000b8a54 terminate_due_to_signal + 148
2   Emacs-x86_64-10_9             	0x00000001000d5e93 emacs_abort + 19
3   Emacs-x86_64-10_9             	0x00000001001a7e2c ns_term_shutdown + 124
4   Emacs-x86_64-10_9             	0x00000001000b8c4b shut_down_emacs + 283
5   Emacs-x86_64-10_9             	0x00000001000b8a19 terminate_due_to_signal + 89
6   Emacs-x86_64-10_9             	0x00000001000d77c6 deliver_fatal_thread_signal + 134
7   libsystem_platform.dylib      	0x00007fff97516eaa _sigtramp + 26
8   ???                           	000000000000000000 0 + 0
9   com.apple.CoreFoundation      	0x00007fff97775102 _CFAutoreleasePoolPop + 50
10  com.apple.Foundation          	0x00007fff944b46c5 -[NSAutoreleasePool release] + 146
11  Emacs-x86_64-10_9             	0x00000001001bfe8c pop_down_menu + 76
12  Emacs-x86_64-10_9             	0x0000000100135f85 unbind_to + 149
13  Emacs-x86_64-10_9             	0x0000000100138256 unwind_to_catch + 102
14  Emacs-x86_64-10_9             	0x00000001001390b4 Fsignal + 1380
15  Emacs-x86_64-10_9             	0x00000001001c1163 -[EmacsDialogPanel runDialogAt:] + 323
16  Emacs-x86_64-10_9             	0x00000001001bfdde ns_popup_dialog + 718
17  Emacs-x86_64-10_9             	0x000000010013a092 Ffuncall + 1202
18  Emacs-x86_64-10_9             	0x000000010017280b exec_byte_code + 2331
19  Emacs-x86_64-10_9             	0x0000000100139e69 Ffuncall + 649
20  Emacs-x86_64-10_9             	0x000000010017280b exec_byte_code + 2331
21  Emacs-x86_64-10_9             	0x0000000100139e69 Ffuncall + 649
22  Emacs-x86_64-10_9             	0x0000000100135eb5 apply1 + 53
23  Emacs-x86_64-10_9             	0x0000000100133ed8 Fcall_interactively + 1272
24  Emacs-x86_64-10_9             	0x000000010013a092 Ffuncall + 1202
25  Emacs-x86_64-10_9             	0x000000010017280b exec_byte_code + 2331
26  Emacs-x86_64-10_9             	0x0000000100139e69 Ffuncall + 649
27  Emacs-x86_64-10_9             	0x000000010013a5ad call1 + 45
28  Emacs-x86_64-10_9             	0x00000001000bcd8e command_loop_1 + 3518
29  Emacs-x86_64-10_9             	0x00000001001387cb internal_condition_case + 251
30  Emacs-x86_64-10_9             	0x00000001000cd85e command_loop_2 + 62
31  Emacs-x86_64-10_9             	0x0000000100138163 internal_catch + 243
32  Emacs-x86_64-10_9             	0x00000001000bb59d recursive_edit_1 + 221
33  Emacs-x86_64-10_9             	0x00000001000bb74c Frecursive_edit + 236
34  Emacs-x86_64-10_9             	0x00000001000ba3a0 main + 5856
35  libdyld.dylib                 	0x00007fff9baf05ad start + 1

Thread 1:: Dispatch queue: com.apple.libdispatch-manager
0   libsystem_kernel.dylib        	0x00007fff96f1eff6 kevent_qos + 10
1   libdispatch.dylib             	0x00007fff95e0c099 _dispatch_mgr_invoke + 216
2   libdispatch.dylib             	0x00007fff95e0bd01 _dispatch_mgr_thread + 52

Thread 2:
0   libsystem_kernel.dylib        	0x00007fff96f1e176 __select + 10
1   Emacs-x86_64-10_9             	0x00000001001a9408 -[EmacsApp fd_handler:] + 264
2   com.apple.Foundation          	0x00007fff94670c6f __NSThread__start__ + 1351
3   libsystem_pthread.dylib       	0x00007fff8be70c13 _pthread_body + 131
4   libsystem_pthread.dylib       	0x00007fff8be70b90 _pthread_start + 168
5   libsystem_pthread.dylib       	0x00007fff8be6e375 thread_start + 13

Thread 3:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	0x00007fff96f18386 mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff96f177c7 mach_msg + 55
2   com.apple.CoreFoundation      	0x00007fff977ad624 __CFRunLoopServiceMachPort + 212
3   com.apple.CoreFoundation      	0x00007fff977acaec __CFRunLoopRun + 1356
4   com.apple.CoreFoundation      	0x00007fff977ac338 CFRunLoopRunSpecific + 296
5   com.apple.AppKit              	0x00007fff99102065 _NSEventThread + 149
6   libsystem_pthread.dylib       	0x00007fff8be70c13 _pthread_body + 131
7   libsystem_pthread.dylib       	0x00007fff8be70b90 _pthread_start + 168
8   libsystem_pthread.dylib       	0x00007fff8be6e375 thread_start + 13

Thread 4:
0   libsystem_kernel.dylib        	0x00007fff96f1e6de __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff8be70729 _pthread_wqthread + 1283
2   libsystem_pthread.dylib       	0x00007fff8be6e365 start_wqthread + 13

Thread 5:: Dispatch queue: NSCGSDisableUpdates
0   libsystem_kernel.dylib        	0x00007fff96f18386 mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff96f177c7 mach_msg + 55
2   com.apple.CoreGraphics        	0x00007fff8fe3ea64 _CGSReenableUpdateForConnection + 92
3   com.apple.CoreGraphics        	0x00007fff8fe3e73d CGSReenableUpdateToken + 532
4   libdispatch.dylib             	0x00007fff95e11871 _dispatch_call_block_and_release + 12
5   libdispatch.dylib             	0x00007fff95e0633f _dispatch_client_callout + 8
6   libdispatch.dylib             	0x00007fff95e0af6f _dispatch_queue_drain + 754
7   libdispatch.dylib             	0x00007fff95e1163b _dispatch_queue_invoke + 549
8   libdispatch.dylib             	0x00007fff95e0633f _dispatch_client_callout + 8
9   libdispatch.dylib             	0x00007fff95e0a1cf _dispatch_root_queue_drain + 1890
10  libdispatch.dylib             	0x00007fff95e09a34 _dispatch_worker_thread3 + 91
11  libsystem_pthread.dylib       	0x00007fff8be7068f _pthread_wqthread + 1129
12  libsystem_pthread.dylib       	0x00007fff8be6e365 start_wqthread + 13

Thread 6:
0   libsystem_kernel.dylib        	0x00007fff96f1e6de __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff8be70729 _pthread_wqthread + 1283
2   libsystem_pthread.dylib       	0x00007fff8be6e365 start_wqthread + 13

Thread 7:
0   libsystem_kernel.dylib        	0x00007fff96f1e6de __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff8be70729 _pthread_wqthread + 1283
2   libsystem_pthread.dylib       	0x00007fff8be6e365 start_wqthread + 13

Thread 8:

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000006  rcx: 0x00007fff5fbfe4a8  rdx: 0x0000000000000000
  rdi: 0x00000000000054ae  rsi: 0x0000000000000006  rbp: 0x00007fff5fbfe4d0  rsp: 0x00007fff5fbfe4a8
   r8: 0x00000000000000ff   r9: 0x000000010131e320  r10: 0x0000000000000000  r11: 0x0000000000000202
  r12: 0x000000010053c680  r13: 0x000000010053c620  r14: 0x0000000000000028  r15: 0x0000000000000020
  rip: 0x00007fff96f1d9fe  rfl: 0x0000000000000202  cr2: 0x00000001093e6000
  
Logical CPU:     0
Error Code:      0x02000025
Trap Number:     133

**Comment by ricardoxvpn** [comment]
+1

**Comment by Porcupine96** [comment]
+1

**Comment by bradyt** [comment]
I'm not using spacemacs, but found this in a web search. Users of `spacemacs` should be taking the trivial step of testing things like this in `emacs -Q`.

I began following the instructions in `M-x report-emacs-bug`. With the suggested build flags, there was no crash, and I realized I had cloned emacs 27. Was going to try building 26, etc, but as it was so slow, and I suspected this was already fixed, I searched and found that the issue was already on the emacs bug tracker archive, at http://lists.gnu.org/archive/html/emacs-bug-tracker/2016-07/msg00210.html.

And searching the repo for that bug#, I believe this is the fixing commit: https://github.com/emacs-mirror/emacs/commit/02ca5db040b57ca4a36c317fcb3fb56f43ab371e.

Should we close this unless someone reproduces with a version after, what, 26.0.91?

@Porcupine96 ^
